### PR TITLE
Fixed namespace propagation after dom manipulation

### LINKED ISSUE #215: Namespace propagation is no working correctly
Sample XML
<o><parent xmlns:foo="bar"><foo:child></foo:child><foo:child></foo:child></parent></o>
1. Parsing an xml fragment where the parent declares a namespace and 2 children contain this namespace.
1. Add the parent's children to the parentParent and remove the parent
1.  serialize the content

Expected: all children should now contain the namespace definition
Actual: only the first child contains the namespace definition

### PR DESCRIPTION
**Pull request by tobiasso85**
When DOM is manipulated the namespace gets correctly propagated to all
children if their namespace was declared in the removed node instead of
only the first child.

This fixes #215 .

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -4,9 +4,14 @@ A JavaScript implementation of W3C DOM for Node.js, Rhino and the browser. Fully
...
 `DOMParser` and `XMLSerializer` interface such as in browser.
 
+About:
+-------
+Fork from [xmldom](https://github.com/jindw/xmldom)
+* fix for namespace propagation after dom manipulation
+
 Install:
 -------
->npm install xmldom
```
Comment (edi9999) [inline readme.md:9]: This shouldn't be in your pull request

File: package.json [hunk 2]
```diff
@@ -4,8 +4,8 @@
   "description": "A W3C Standard XML DOM(Level2 CORE) implementation and parser(DOMParser/XMLSerializer).",
   "keywords": ["w3c","dom","xml","parser","javascript","DOMParser","XMLSerializer"],
   "author": "jindw <jindw@xidea.org> (http://www.xidea.org)",
-  "homepage": "https://github.com/jindw/xmldom",
```
Comment (edi9999) [inline package.json:7]: This shouldn't be in your PR

File: dom.js [hunk 3]
```diff
@@ -1051,9 +1051,9 @@ function serializeToString(node,buf,isHTML,nodeFilter,visibleNamespaces){
 		}
 		return;
 	case ATTRIBUTE_NODE:
-		return buf.push(' ',node.name,'="',node.value.replace(/[<&"]/g,_xmlEncoder),'"');
+		return buf.push(' ',node.name,'="',String(node.value).replace(/[<&"]/g,_xmlEncoder),'"');
```
Comment (edi9999) [inline dom.js:1054]: Why did you add String ?

File: test/dom/serializer.js [hunk 4]
```diff
@@ -1,15 +1,51 @@
...
-    console.assert(doc.documentElement.firstChild.toString() == '<script></script>');
-  },
+	'text node containing "]]>"': function() {
+		var doc = new DOMParser().parseFromString('<test/>', 'text/xml');
+		doc.documentElement.appendChild(doc.createTextNode('hello ]]> there'));
+		assert.equal(doc.documentElement.firstChild.toString(), 'hello ]]> there', doc.documentElement.firstChild.toString());
+	},
+	'<script> element with no children': function() {
+		var doc = new DOMParser({xmlns: {xmlns: 'http://www.w3.org/1999/xhtml'}}).parseFromString('<html2><script></script></html2>', 'text/html');
+		//console.log(doc.documentElement.firstChild.toString(true))
```
Comment (edi9999) [inline test/dom/serializer.js:15]: Can you remove this console.log ?

**Review by edi9999 — CHANGES_REQUESTED** [review]

**Comment by tobiasso85** [comment]
continued pull request with #221 to have it completely separate

**Pull request closed without merging**